index.js

```diff
@@ -57,21 +57,29 @@ function getTransaction(env, tablesToLock_) {
 
 function withQueryRowsAsync(client) {
   return Object.assign(client, {
-    queryRowsAsync: (query, args) => client.queryAsync(query, args).then(getRows)
+    queryRowsAsync: (query, args) => queryAsync(client, query, args).then(getRows)
   })
 }
 
 function queryRowsAsync(env, query, args) {
   var argsArray = args || []
   return using(getConnection(env), function (connection) {
-    return connection.queryAsync(query, argsArray)
+    return queryAsync(connection, query, argsArray)
   }).then(getRows)
 }
 
 function getRows(res) {
   return res.rows
 }
 
+function queryAsync(client, query, args) {
+  if (query.values && args && args.length) {
```

Comment:
Could the check be more specific for the types of query.values and for the type of args?